### PR TITLE
[mlir][vector] Refactor linearize.mlir

### PR DESCRIPTION
**Pull request by banach-space**
This patch refactors the `linearize.mlir` test - currently it contains
some duplication and can be tricky to follow.

Summary of changes:
  * reduce duplication by introducing a shared check prefix (`ALL`) and
    by introducing `-check-prefixes`,
  * make sure that every "check" line is directly above the
    corresponding line of input MLIR,
  * group check lines corresponding to a particular prefix together (so
    that it's easier to see the expected output for a particular
    prefix),
  * remove `CHECK` from prefix names (with multiple prefixes that's just
    noise that can be avoided) and use a bit more descriptive prefixes
    instead (`CHECK0` -> `BW-0`, where `BW` stands for bitwidth),
  * unify indentation,
  * `nonvec_result` -> `test_tensor_no_linearize` (for consistency with
    `test_index_no_linearize`).

NOTE: This change only updates the format of the "CHECK" lines and
doesn't affect what's being tested.

This change is intended as preparation for adding support for scalable
vectors to `LinearizeConstant` and `LinearizeVectorizable` - i.e.
patterns that `linearlize.mlir` is meant to test.

### CODE REVIEW

**Review by Hardcode84 — APPROVED** [review]
LGTM, Thanks

File: mlir/test/Dialect/Vector/linearize.mlir [hunk 1]
```diff
@@ -1,92 +1,89 @@
...
-// RUN: mlir-opt %s -split-input-file -test-vector-linearize=target-vector-bitwidth=0 | FileCheck %s --check-prefix=CHECK0
-
-// CHECK-LABEL: test_linearize
-// CHECK128-LABEL: test_linearize
-// CHECK0-LABEL: test_linearize
-//  CHECK-SAME: (%[[ORIG_ARG:.*]]: vector<2x2xf32>)
-//  CHECK128-SAME: (%[[ORIG_ARG:.*]]: vector<2x2xf32>)
-//       CHECK: %[[ARG:.*]] = vector.shape_cast %[[ORIG_ARG]] : vector<2x2xf32> to vector<4xf32>
-//       CHECK128: %[[ARG:.*]] = vector.shape_cast %[[ORIG_ARG]] : vector<2x2xf32> to vector<4xf32>
+// RUN: mlir-opt %s -split-input-file -test-vector-linearize | FileCheck %s --check-prefixes=ALL,DEFAULT
```
Comment (bviyer) [inline mlir/test/Dialect/Vector/linearize.mlir:1]: Why did you add a new file-prefix instead of keeping the default one (by default I meant just using "CHECK" not the "DEFAULT" tag)?

File: mlir/test/Dialect/Vector/linearize.mlir [hunk 2]
```diff
@@ -95,10 +92,11 @@ func.func @test_index_no_linearize(%arg0: vector<2x2xindex>, %arg1: vector<2x2xi
 
 // vectorizable operation (arith.mulf) with tensor result types.
 
-func.func @nonvec_result(%arg0: tensor<2x2xf32>, %arg1: tensor<2x2xf32>) -> (tensor<2x2xf32>, tensor<2x2xf32>) {
-    // CHECK: %[[MULF:.*]] = arith.mulf %arg0, %arg1 : tensor<2x2xf32>
-    // CHECK128: %[[MULF:.*]] = arith.mulf %arg0, %arg1 : tensor<2x2xf32>
-    // CHECK0: %[[MULF:.*]] = arith.mulf %arg0, %arg1 : tensor<2x2xf32>
+// ALL-LABEL: test_tensor_no_linearize
+func.func @test_tensor_no_linearize(%arg0: tensor<2x2xf32>, %arg1: tensor<2x2xf32>) -> (tensor<2x2xf32>, tensor<2x2xf32>) {
+    // DEFAULT: %[[MULF:.*]] = arith.mulf %arg0, %arg1 : tensor<2x2xf32>
```
Comment (bviyer) [inline mlir/test/Dialect/Vector/linearize.mlir:97]: Can't you replace all these 3 with ALL?

File: mlir/test/Dialect/Vector/linearize.mlir (hunk 1 above)
Comment (banach-space) [reply mlir/test/Dialect/Vector/linearize.mlir:1]: `DEFAULT` documents the fact that the _default config_ is being tested. And `-test-vector-linearize` means "default". When using `CHECK`, we are missing an opportunity to document what configuration is being tested.

I appreciate that it's a matter of style/preference and both approaches are 100% valid. Happy to revert this change if that's your preference.

File: mlir/test/Dialect/Vector/linearize.mlir (hunk 2 above)
Comment (banach-space) [reply mlir/test/Dialect/Vector/linearize.mlir:97]: Oh, I missed that, thanks!

File: mlir/test/Dialect/Vector/linearize.mlir (hunk 1 above)
Comment (bviyer) [reply mlir/test/Dialect/Vector/linearize.mlir:1]: No need to revert. Just asking for my knowledge :).

**Review by bviyer — APPROVED** [review]
